src/main/kotlin/com/exactpro/th2/check1/rule/CheckTaskUtils.kt

```diff
@@ -0,0 +1,101 @@
+ */
+@file:JvmName("CheckTaskUtils")
+package com.exactpro.th2.check1.rule
+
+import com.exactpro.th2.common.event.Event
+import com.exactpro.th2.common.event.EventUtils
+import com.exactpro.th2.common.grpc.EventBatch
+import com.exactpro.th2.common.grpc.EventBatch.Builder
+import com.exactpro.th2.common.grpc.EventID
+import com.exactpro.th2.common.grpc.EventStatus
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.google.protobuf.ByteString
+import com.google.protobuf.TextFormat
+import mu.KotlinLogging
+import com.exactpro.th2.common.grpc.Event as ProtoEvent
+
+val OBJECT_MAPPER = ObjectMapper()
+val LOGGER = KotlinLogging.logger {}
+
+fun Event.disperseToBatches(maxEventBatchContentSize: Int, parentEventId: EventID): List<EventBatch> {
+    require(maxEventBatchContentSize > 0) {
+        "'maxEventBatchContentSize' should be greater than zero, actual: $maxEventBatchContentSize"
+    }
+
+    val events = toListProto(parentEventId)
+
+    val result = ArrayList<EventBatch>()
+    val eventGroups = events.groupBy(ProtoEvent::getParentId)
+    batch(maxEventBatchContentSize, result, eventGroups, parentEventId)
+    return result
+}
+
+private fun batch(maxEventBatchContentSize: Int, result: MutableList<EventBatch>, eventGroups: Map<EventID, List<ProtoEvent>>, eventID: EventID) {
+    val factory = { EventBatch.newBuilder().setParentEventId(eventID) }
+
+    var builder = factory.invoke()
+
+    checkNotNull(eventGroups[eventID]) {
+        "Neither of events refers to ${TextFormat.shortDebugString(eventID)}"
+    }.forEach { event ->
+        val checkedEvent = checkAndRebuild(maxEventBatchContentSize, event)
+
+        LOGGER.trace("Process ${checkedEvent.name} ${checkedEvent.type}")
+        if(eventGroups.containsKey(checkedEvent.id)) {
+            result.add(checkAndBuild(maxEventBatchContentSize, factory.invoke()
+                .addEvents(checkedEvent)))
+
+            batch(maxEventBatchContentSize, result, eventGroups, checkedEvent.id)
+        } else {
+            if(builder.eventsCount > 0
+                && builder.getContentSize() + checkedEvent.getContentSize() > maxEventBatchContentSize) {
+                result.add(checkAndBuild(maxEventBatchContentSize, builder))
+                builder = factory.invoke()
+            }
+            builder.addEvents(checkedEvent)
+        }
+    }
+
+    if(builder.eventsCount > 0) {
+        result.add(checkAndBuild(maxEventBatchContentSize, builder))
+    }
+}
+
+private fun checkAndBuild(maxEventBatchContentSize: Int, builder: Builder): EventBatch {
+    val contentSize = builder.getContentSize()
+    check(contentSize <= maxEventBatchContentSize) {
+        "The smallest batch size exceeds the max event batch content size, max $maxEventBatchContentSize, actual $contentSize"
+    }
+
+    return builder.build()
+}
+
+private fun checkAndRebuild(maxEventBatchContentSize: Int, event: ProtoEvent): ProtoEvent {
+    return if (event.body.size() > maxEventBatchContentSize) {
```

Comment:
I think we need to use `ProtoEvent.getContentSize()` for consistency. It might confuse when we use body.size() in one place and getContentSize() in another